Validator: Support VK_KHR_shader_float_controls

AttilioProvenzano-ARM opened the pull request:
Adds validator options to specify enabled float controls features.

By default, none of the features are considered to be active, so
validation will fail if any float controls are used.

For validator options that match directly to float control features,
the value is a bitwise OR of supported bit widths, e.g. (16 | 32).
The options for separate denorm and rounding modes are simple booleans.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/SPIRV-Tools?pullRequest=2177) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/SPIRV-Tools?pullRequest=2177) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/SPIRV-Tools?pullRequest=2177) it.</sub>

dneto0 commented:
Thanks for filing this.  There's a debate about how to split responsibilities between the Vulkan validation layers and SPIR-V validation.  I'll start an email discussion with the layers maintainers.
cc: @alan-baker @chrisforbes @mark-lunarg

dneto0 commented:
Offline discussion concluded these checks are better done in the layers.  See https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/674 

I'll keep this open until resolved there.

AttilioProvenzano-ARM commented:
Sorry about the delay, it was a busy time with GDC preparations and I couldn't look at the layers yet. I'll aim at doing that right after GDC. :)

AttilioProvenzano-ARM commented:
@dneto0 I've updated the PR for the layers, I've implemented the same checks there: https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/513 if you want to have a look :)

AttilioProvenzano-ARM commented:
I believe this can be closed now, as the changes were implemented in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/513, after the discussion in https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/674.
